### ISSUE TITLE
[inductor][cpu]GPTNeoForSequenceClassification AMP single/multiple thread static/dynamic shape default/cpp accuracy failure

### ISSUE THREAD

**Issue by zxd1997066** [pytorch/pytorch#135689]
### 🐛 Describe the bug

GPTNeoForSequenceClassification AMP single/multiple thread static/dynamic shape default/cpp accuracy failure in 2024-09-07 nightly release
```
loading model: 0it [00:21, ?it/s]cpu  eval  GPTNeoForSequenceClassification    

WARNING:common:fp64 golden ref were not generated for GPTNeoForSequenceClassification. Setting accuracy check to cosine
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
W0910 22:42:04.897982 129817 torch/_dynamo/utils.py:1719] Similarity score=0.9893215894699097
E0910 22:42:04.898772 129817 torch/_dynamo/utils.py:1670] Accuracy failed for key name logits
WARNING:common:Trying to call the empty_gpu_cache for device: cpu, which is not in list [cuda, xpu]
fail_accuracy
dev,name,batch_size,accuracy,calls_captured,unique_graphs,graph_breaks,unique_graph_breaks,autograd_captures,autograd_compiles,cudagraph_skips
cpu,GPTNeoForSequenceClassification,1,fail_accuracy,1040,3,2,1,0,0,2
```

### Versions

</table><p>SW info</p><table border="1" class="dataframe table">
  <thead>
    <tr style="text-align: right;">
      <th>name</th>
      <th>target_branch</th>
      <th>target_commit</th>
      <th>refer_branch</th>
      <th>refer_commit</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>torchbench</td>
      <td>main</td>
      <td>23512dbe</td>
      <td>main</td>
      <td>23512dbe</td>
    </tr>
    <tr>
      <td>torch</td>
      <td>main</td>
      <td>3bebc09be9845c0779f190489e8d4caa9e2653c8</td>
      <td>main</td>
      <td>c140fa1426603322a5a69ef91300f13489db5970</td>
    </tr>
    <tr>
      <td>torchvision</td>
      <td>main</td>
      <td>0.19.0a0+d23a6e1</td>
      <td>main</td>
      <td>0.19.0a0+d23a6e1</td>
    </tr>
    <tr>
      <td>torchtext</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
      <td>main</td>
      <td>0.16.0a0+b0ebddc</td>
    </tr>
    <tr>
      <td>torchaudio</td>
      <td>main</td>
      <td>2.5.0a0+97ed7b3</td>
      <td>main</td>
      <td>2.5.0a0+97ed7b3</td>
    </tr>
    <tr>
      <td>torchdata</td>
      <td>main</td>
      <td>0.7.0a0+11bb5b8</td>
      <td>main</td>
      <td>0.7.0a0+11bb5b8</td>
    </tr>
    <tr>
      <td>dynamo_benchmarks</td>
      <td>main</td>
      <td>nightly</td>
      <td>main</td>
      <td>nightly</td>
    </tr>
  </tbody>
</table>

</table>

Repro:
[inductor_single_run.sh](https://github.com/chuanqi129/inductor-tools/blob/main/scripts/modelbench/inductor_single_run.sh)
bash inductor_single_run.sh multiple inference accuracy huggingface GPTNeoForSequenceClassification amp first static cpp
Suspected guilty commit: https://github.com/pytorch/pytorch/commit/52c7c89ea486f8124b740ad6d2ee055812e913ab
[huggingface-GPTNeoForSequenceClassification-inference-amp-dynamic-default-single-accuracy-crash_guilty_commit.log](https://github.com/user-attachments/files/16961940/huggingface-GPTNeoForSequenceClassification-inference-amp-dynamic-default-single-accuracy-crash_guilty_commit.log)
cc @WeizhuoZhang-intel @chuanqi129

**Comment by leslie-fang-intel** [comment]
cc @yuchengliu1 to take a look
